app/enterprise/0.32-x/upgrades-migrations.md

```diff
@@ -0,0 +1,56 @@
+13. Run the `kong migrations up` command and start Kong.
+
+Once a new version is available, you are informed by a distribution list email or via your designated Customer Success Engineer (Premium subscription only). The newer version will be available in the same repository in Bintray.
+
+Although you may already have backup processes in place, a most recent backup of your database prior upgrade is needed due to the fact that represents the most recent versions of your entities and other metrics which need to be preserved while upgrading. After such backup any changes applied via the Admin API won't appear in the destination environment, therefore we suggest not making any modifications on the Admin API during the upgrade process.
+ For database backups you can use specific tools for each database such as `pg_dump` for Postgres or `nodetool snapshot` for Cassandra; however we recommend to always refer to an experienced DBA.
+
+Once you downloaded the package you review the change log to understand whether anything in your infrastructure needs to be upgraded consequentially, particular attention must be made around the supported database versions which may fall into a “no more supported state” due addition which Kong developers leverage to make queries and operation more efficient in the DAO and database level.
+Installing the newer package updates core libraries, plugins including a `kong.conf.default` which you can review to take into account newer or changes to configuration properties, those are usually documented in the changelog as well.
+Following the changelog itself is essential that you review any breaking changes and adapt your deployment process, kong configurations files, custom nginx template and anything else may be affected.
+
+**When performing upgrades in a solution with multiple kong nodes running, always shut down the other nodes and perform the upgrade operation one node at the time.**
+
+Running the `kong migrations up` command an update of the existing schema to the newer version is performed, old kong packages running on updated database schemas are not guaranteed to work correctly.
+At this stage, you want to validate the upgrade and smoke test your preproduction environment.
+One smart way of doing so is verifying the behaviour with the application of the canary plugin on a low priority api/service entity in preproduction redirecting it to correspondent entity in the upgraded production (if you setup allows doing so) and you may want to add also request termination plugin in the source environment, or doing so entirely with a reverse-proxy hard redirect for specifics low priority routes, if you have any in your infrastructure, nevertheless using the old school postman or curl commands.
+Once preproduction is validated you can go ahead and perform the same upgrade in production according to with your experience in preproduction following your business processes.
```

Comment:
"Once preproduction is validated you can go ahead and perform the same upgrade in production according to with your experience in preproduction following your business processes." -> "Once preproduction is validated, you can go ahead and perform the same upgrade in production" - I'm not sure what the "according to with your experience in preproduction following your business processes" part really means, and also it is incorrect English, but I'm having a hard time suggesting a correction because I'm not sure what you are trying to say :)

Reply:
I meant that you might need to adjust your infrastructure and adapt your business process if the upgrade requires, users will find that out whilst upgrading pre-production; Grammarly says "according to with your experience in preproduction following your business processes" is valid English.
Users' business processes and infrastructure adaptation is out of scope.

Reply:
Ha ha, computerized grammar checking is far more difficult to get precisely right than computerized spelling checking. In this case "according to with your experience in preproduction following your business processes" is not correct grammar, or at least if it is correct, it is so unclear as to be not understandable. Can you add some of the words you mention in your comment https://github.com/Kong/docs.konghq.com/pull/774#discussion_r213231021 ? You wrote it nice a clear in that comment!